crates/nu-cli/src/util.rs

```diff
@@ -199,17 +204,126 @@ fn gather_env_vars(
     }
 }
 
-pub fn eval_source(
+pub fn make_main_call(
+    engine_state: &mut EngineState,
```

Comment:
I think it's more conventional for us to always take a `&mut StateWorkingSet`, just in case there's something more you want to do with it. It works in this case but it would probably be good to follow that.